From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Patrick McCarty <patrick.mccarty@intel.com>
Date: Tue, 6 Aug 2019 12:30:59 -0700
Subject: [PATCH] Whitelist syscall futex() for seccomp

Linking to libpthread directly has the side-effect of futex() getting
called, so it needs to be whitelisted.

Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 src/seccomp.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/seccomp.c b/src/seccomp.c
index 1b9d9b8..54ea144 100644
--- a/src/seccomp.c
+++ b/src/seccomp.c
@@ -167,6 +167,7 @@ enable_sandbox_full(void)
  	ALLOW_RULE(fcntl64);
 	ALLOW_RULE(fstat);
  	ALLOW_RULE(fstat64);
+	ALLOW_RULE(futex);
 	ALLOW_RULE(getdents);
 #ifdef __NR_getdents64
 	ALLOW_RULE(getdents64);
